core/app/models/spree/product.rb

```diff
@@ -650,7 +650,7 @@ def to_csv(store = nil)
       csv_lines = []
 
       if has_variants?
-        variants.each_with_index do |variant, index|
+        variants_including_master.each_with_index do |variant, index|
```

Comment:
This will list the master variant as the first row of all variants and all the common data like taxonomies, images, etc. will be exported to that row